frontend/src/features/properties/common/slices/propertyNameSlice.ts

```diff
@@ -0,0 +1,23 @@
+import { createSlice, createAction } from '@reduxjs/toolkit';
+
+export const savePropertyNames = createAction<String[]>('savePropertyNames');
+export const clearPropertyNames = createAction('clearPropertyNames');
+/**
+ * Slice to handle storage of property names
+ */
+const propertyNameSlice = createSlice({
+  name: 'propertyNames',
+  initialState: [] as String[],
+  reducers: {},
+  extraReducers: (builder: any) => {
+    builder.addCase(savePropertyNames, (state: any, action: any) => {
+      console.log(action.payload);
```

Comment:
What's a slice/what is this doing? Also there's still a console.log here

Reply:
It creates the action creators and action types based off the reducers and state (bunch of redux stuff). It is part of the Redux Toolkit that cleans up a lot of the normal work you would have to do!